rmw_fastrtps_shared_cpp/src/rmw_node.cpp

```diff
@@ -191,29 +216,23 @@ get_security_file_paths(
   };
   size_t num_files = sizeof(file_names) / sizeof(char *);
 
-  std::string file_prefix("file://");
-
   for (size_t i = 0; i < num_files; i++) {
-    rcutils_allocator_t allocator = rcutils_get_default_allocator();
-    char * file_path = rcutils_join_path(node_secure_root, file_names[i], allocator);
-
-    if (!file_path) {
-      return false;
-    }
 
-    if (rcutils_is_readable(file_path)) {
-      security_files_paths[i] = file_prefix + std::string(file_path);
-    } else {
-      allocator.deallocate(file_path, allocator.state);
+    if (!get_security_file_path(security_files_paths[i], file_names[i], node_secure_root)) {
       return false;
     }
-
-    allocator.deallocate(file_path, allocator.state);
   }
 
   return true;
 }
 
+bool
+load_security_logging_options(std::string& distribute, std::string& event_log_level,
+                              std::string& log_file, const std::string& security_logging_file_path) {
```

Comment:
Right, I have not included the QoS options just yet. At the moment, this PR is definitely a draft to illustrate the design more than a full fledged implementation. We want to make sure we have the pieces fitting together before Foxy API freeze.

Concerning the aggregation of the parameters in a struct, given that the function is only use internally here, I don't see a need for it. Moreover, I don't want to give the impression that such struct is somewhat a duplicate of `rmw_node_security_options_t`, thus that they should be merged. We chose to implement the security logging parameters at this layer precisely to avoid modifications at `rcl` level.